profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PprofAllocationEventExporter.java

```diff
@@ -69,21 +71,27 @@ public void export(RecordedEvent event, AllocationEventSampler sampler, SpanCont
 
     Sample.Builder sample = Sample.newBuilder();
     sample.addValue(allocationSize);
-    // XXX StackSerializer limits stack depth. Although I believe jfr also limits it should we
-    // attempt to limit the depth here too?
-    for (RecordedFrame frame : stackTrace.getFrames()) {
-      RecordedMethod method = frame.getMethod();
-      if (method == null) {
-        sample.addLocationId(pprof.getLocationId("unknown", "unknown.unknown", -1));
-      } else {
-        sample.addLocationId(
-            pprof.getLocationId(
-                "unknown", // file name is not known
-                method.getType().getName() + "." + method.getName(),
-                frame.getLineNumber()));
-      }
+
+    if (stackTrace.isTruncated() || stackTrace.getFrames().size() > stackDepth) {
+      pprof.addLabel(sample, THREAD_STACK_TRUNCATED, true);
     }
 
+    stackTrace.getFrames().stream()
+        .limit(stackDepth)
+        .forEachOrdered(
+            frame -> {
+              RecordedMethod method = frame.getMethod();
+              if (method == null) {
+                sample.addLocationId(pprof.getLocationId("unknown", "unknown.unknown", -1));
+              } else {
+                sample.addLocationId(
+                    pprof.getLocationId(
+                        "unknown", // file name is not known
+                        method.getType().getName() + "." + method.getName(),
+                        frame.getLineNumber()));
+              }
+            });
```

Comment:
This logic is present for pprof allocation exporter, but not for pprof cpu exporter (or any text exporters). Maybe add a bunch of comments to the exporters that rely on other classes to do the truncation? Or somehow unify it.

Reply:
for pprof cpu truncation is done in `StackTraceParser`, for text allocation in `StackSerializer`. The only once that does not truncate is text cpu where truncating would require parsing & rewriting the stacks

Reply:
I see. In case of text+cpu, do we expect that JFR will respect the limit that was passed along to it? (except when the user overrides it)?

Reply:
no, I would not expect the stack depth limit to apply to thread dump events.

Reply:
Alright, then are the following correct (as of this current PR)?

1. **text/cpu**: the truncation option is ignored
2. **text/allocation**: we try to ask JFR to truncate and also truncate on our own if it's still too tall
3. **pprof/cpu**: we truncate on our own
4. **pprof/allocation**: we try to ask JFR to truncate and also truncate on our own if it's still too tall

Reply:
yes